Project1/Project1/Groot.cpp

```
@@ -0,0 +1,61 @@
+#include <cmath>
+#include <iostream>
+
+enum return_values
+{
+    SUCCESS = 0,
+    FAILURE = -1,
+};
+
+float print_sqrt(float num)
+{
+    // A function to print and return a given number, throwing error if occurs and return 1-
+    float sqrt = 0;
+    try
+    {
+        sqrt = std::sqrt(num); // calculating sqrt of the num
+        std::cout << "sqrt of input is:  " << sqrt;
+        return sqrt;
+    }
+    catch (...)
+    {
+        throw "Error while taking sqrt from number, please enter a positive number";
+    }
+}
+
+float get_input_num(){
```

use exception instead of returning error value

---

you didn't do this

---

What are you mean? I did throw an exception. Do I need to throw the program exception and not mine massege?

---

What are you mean? I did throw an exception. Do I need to throw the program exception and not mine massege?

---

you return your number if the input is valid and an error value if its not valid. instead of returning an error value you can throw an exception, and it will be easier to use your function. instead of using complex checks(which might not seem complex now but once there are many different errors there might be)